pkg/kubelet/kubelet_pods.go

```diff
@@ -145,10 +145,10 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 	// - container is not an infrastructure (pause) container
 	// - container is not already mounting on /etc/hosts
 	// - OS is not Windows
-	// - if it is Windows, ContainerD is used.
 	// Kubernetes will not mount /etc/hosts if:
 	// - when the Pod sandbox is being created, its IP is still unknown. Hence, PodIP will not have been set.
-	mountEtcHostsFile := len(podIPs) > 0 && supportsSingleFileMapping
+	mountEtcHostsFile := len(podIPs) > 0 && supportsSingleFileMapping && runtime.GOOS != "windows"
```

Comment:
Containerd on Windows supports single file mapping (https://github.com/kubernetes/kubernetes/pull/83730/files) as of 1.21.  We may want to skip it here:  https://github.com/kubernetes/kubernetes/blob/2e93b3924ae6eb2b6a2b914fe400a006bd676308/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L282-L285

I think a HostProcess that uses the host network would inherit the host's `hosts` file, since it doesn't have it's own file system?  @dcantah  I am not sure that if we mapped it in that it would be useful in any way meaningful?

Reply:
It's not useful you're correct

Reply:
I think the fix here is actually just to skip mounting it if it's a hostprocess container

Reply:
@brasmith-ms fyi this seems like something that should be mentioned in the docs

Reply:
Once again, I think we make the new condition that we check here if the container is a host process container instead of if it's not a windows container. This is likely here for a reason for regular Windows server containers.